shell/platform/darwin/macos/framework/Source/FlutterEmbedderExternalTextureUnittests.mm

```diff
@@ -18,6 +19,47 @@
 #include "third_party/skia/include/core/SkSamplingOptions.h"
 #include "third_party/skia/include/core/SkSurface.h"
 
+@interface TestExternalTexture : NSObject <FlutterTexture>
+
+- (nonnull instancetype)initWidth:(size_t)width height:(size_t)height;
+
+@end
+
+@implementation TestExternalTexture {
+  size_t _width;
+  size_t _height;
+}
+
+- (nonnull instancetype)initWidth:(size_t)width height:(size_t)height {
+  if (self = [super init]) {
+    _width = width;
+    _height = height;
+  }
+  return self;
+}
+
+- (CVPixelBufferRef)copyPixelBuffer {
+  return [self pixelBuffer];
+}
+
+- (CVPixelBufferRef)pixelBuffer {
+  NSDictionary* options = @{
+    // This key is required to generate SKPicture with CVPixelBufferRef in metal.
+    (NSString*)kCVPixelBufferMetalCompatibilityKey : @YES
+  };
+  CVPixelBufferRef pxbuffer = NULL;
+  CVReturn status =
+      CVPixelBufferCreate(kCFAllocatorDefault, _width, _width, kCVPixelFormatType_32BGRA,
+                          (__bridge CFDictionaryRef)options, &pxbuffer);
+
+  NSParameterAssert(status == kCVReturnSuccess && pxbuffer != NULL);
```

Comment:
You aren't checking parameters of this method. So I think `NSAssert` makes more sense.